app/src/main/java/tv/remo/android/controller/activities/ExternalControlActivity.kt

```diff
@@ -0,0 +1,196 @@
+                apiKeyIsInvalid = invalidKey
+                binding.channelNameText.text = getString(R.string.channelNameString, channelName)
+            }
+        }
+        binding.startStreamButton.setOnClickListener(this::onStartStreamButtonClicked)
+
+        // There are several reasons that we don't need the security of making the user approve
+        // starting the stream on the REV Robotics Control Hub, and since that device is headless,
+        // we can just immediately pretend that the start stream button was clicked.
+        if (Build.MANUFACTURER == "REV Robotics" && Build.MODEL.contains("Control Hub")) {
+            onStartStreamButtonClicked(binding.startStreamButton)
+        }
+    }
+
+    private fun processExternalAppParameters(): ExternalAppParameters {
+        return ExternalAppParameters(
+            apiKey = intent.getStringExtra("ApiKey")?: "fakeKey",
+            enableRobot = intent.getBooleanExtra("EnableRobot", true),
+            robotCommunicationDriver = when(intent.getStringExtra("RobotCommunicationDriver")) {
+                "BluetoothClassic" -> BluetoothClassicDriver::class.java
+                "FelhrUsb" -> FelhrUsbSerialDriver::class.java
+                "RemoBroadcaster" -> RemoBroadcaster::class.java
+                else -> RemoBroadcaster::class.java
+            },
+            robotProtocolTranslator = when (intent.getStringExtra("RobotProtocolTranslator")) {
+                "Arduino" -> ArduinoTranslator::class.java
+                "ArduinoSendSingleChar" -> ArduinoSendSingleCharTranslator::class.java
+                "NXTJoystick" -> NXTJoystickDriverTranslator::class.java
+                "SingleByte" -> SingleByteTranslator::class.java
+                else -> ArduinoTranslator::class.java
+            },
+            enableCamera = intent.getBooleanExtra("EnableCamera", false),
+            cameraWidth = intent.getIntExtra("CameraWidth", 640),
+            cameraHeight = intent.getIntExtra("CameraWidth", 480),
+            cameraOrientation = when(intent.getStringExtra("CameraOrientation")) {
+                "DIR_0" -> Orientation.DIR_0
+                "DIR_90" -> Orientation.DIR_90
+                "DIR_180" -> Orientation.DIR_180
+                "DIR_270" -> Orientation.DIR_270
+                else -> Orientation.DIR_90
+            },
+            cameraDeviceId = intent.getIntExtra("CameraDeviceId", -1),
+            cameraBitrateKbps = intent.getIntExtra("CameraBitrateKbps", 1024),
+            enableMicrophone = intent.getBooleanExtra("EnableMicrophone", false),
+            microphoneVolumeMultiplier = intent.getDoubleExtra("MicrophoneVolumeMultiplier", 1.0),
+            microphoneBitrateKbps = intent.getIntExtra("MicrophoneBitrateKbps", 64))
+    }
+
+    private fun applyExternalAppParameters(externalAppParameters: ExternalAppParameters) {
+        // TODO(Noah): Somehow one of these causes the camera settings activity to crash, but I don't see any discrepancies
```

Comment:
@NoahAndrews This error has been fixed via https://github.com/remotv/controller-for-android/commit/a30a2c1e3d1fb8c3304d6f9988566c7e88cbf439

The issue was it was saving to preferences as an Int, but the ListPreference can only handle Strings, so it crashed. Only issue currently with this fix is to remove different crashing, the app data has to be cleared to remove that Int pref. Not sure if I should make it call `getInt` if `getString` fails, or leave the getter the same as it was before the fix, which first tried to `getInt`, then tried `getString` if the first call failed, which it apparently always did

Reply:
Changing the type of an existing preference is a recipe for bugs IMO. I would make a preference with a new name, and then migrate the data from the old preference to the new one in application (not activity) start.  Once it's been migrated you can delete the data in the old preference. It also helps to rename the string with the old preference key to clarify that it is the legacy version.